pcsx2/gui/Dialogs/GSDumpDialog.cpp

```diff
@@ -0,0 +1,806 @@
+	int i = 0, h = 0, j = 0;
+	m_dump_list->AppendColumn("Dumps");
+	// set the column size to be exactly of the size of our list
+	m_dump_list->GetSize(&h, &j);
+	m_dump_list->SetColumnWidth(0, h);
+
+	while (cont)
+	{
+		m_dump_list->InsertItem(i, filename.substr(0, filename.find_last_of(".")));
+		i++;
+		cont = snaps.GetNext(&filename);
+	}
+}
+
+void Dialogs::GSDumpDialog::SelectedDump(wxListEvent& evt)
+{
+	wxString filename_preview = g_Conf->Folders.Snapshots.ToAscii() + ("/" + evt.GetText()) + ".png";
+	wxString filename = g_Conf->Folders.Snapshots.ToAscii() + ("/" + evt.GetText()) + ".gs";
+	if (wxFileExists(filename_preview))
+	{
+		auto img = wxImage(filename_preview);
+		img.Rescale(250 * MSW_GetDPIScale(), 200 * MSW_GetDPIScale(), wxIMAGE_QUALITY_HIGH);
+		m_preview_image->SetBitmap(wxBitmap(img));
+		delete m_selected_dump;
+		m_selected_dump = new wxString(filename);
+	}
+	else
+	{
+		m_preview_image->SetBitmap(EmbeddedImage<res_NoIcon>().Get());
+	}
+}
+
+void Dialogs::GSDumpDialog::RunDump(wxCommandEvent& event)
+{
+	m_debug_mode->Enable();
+	m_start->Enable();
+	m_step->Enable();
+	m_selection->Enable();
+	m_vsync->Enable();
+	GetCorePlugins().Shutdown();
+	m_thread->Start();
+	return;
+}
+
+void Dialogs::GSDumpDialog::ProcessDumpEvent(GSData event, char* regs)
+{
+    switch (event.id)
+	{
+		case Transfer:
+		{
+			switch (event.path)
+			{
+				case Path1Old:
+				{
+					u32* data = (u32*)malloc(16384);
+					int addr = 16384 - event.length;
+					memcpy(data, event.data + addr, event.length);
+					GSgifTransfer1(data, addr);
+					free(data);
+					break;
+				}
+				case Path1New:
+				{
+					GSgifTransfer((u32*)event.data, event.length / 16);
+					break;
+				}
+				case Path2:
+				{
+					GSgifTransfer2((u32*)event.data, event.length / 16);
+					break;
+				}
+				case Path3:
+				{
+					GSgifTransfer3((u32*)event.data, event.length / 16);
+					break;
+				}
+				default:
+					break;
+			}
+			break;
+		}
+		case VSync:
+		{
+			GSvsync((*((int*)(regs + 4096)) & 0x2000) > 0 ? (u8)1 : (u8)0);
+			break;
+		}
+		case ReadFIFO2:
+		{
+			u64* arr = (u64*)malloc(*((int*)event.data));
+			GSreadFIFO2(arr, *((int*)event.data));
+			free(arr);
+			break;
+		}
+		case Registers:
+		{	
+			memcpy(regs, event.data, 8192);
+			break;
+		}
+	}
+}
+
+void Dialogs::GSDumpDialog::StepPacket(wxCommandEvent& event)
+{
+	m_button_events.push_back(GSEvent{Step, 0});
+}
+
+void Dialogs::GSDumpDialog::ToCursor(wxCommandEvent& event)
+{
+	m_button_events.push_back(GSEvent{RunCursor, wxAtoi(m_gif_list->GetItemText(m_gif_list->GetFocusedItem()).BeforeFirst('-'))});
+}
+
+void Dialogs::GSDumpDialog::ToVSync(wxCommandEvent& event)
+{
+	m_button_events.push_back(GSEvent{RunVSync, 0});
+}
+
+void Dialogs::GSDumpDialog::ToStart(wxCommandEvent& event)
+{
+	m_button_events.push_back(GSEvent{RunCursor, 0});
+}
+
+void Dialogs::GSDumpDialog::GenPacketList(std::vector<GSData>& dump)
+{
+	int i = 0;
```

Comment:
Maybe move it inside the loop since it's only used inside?

Reply:
Yes it is only used inside because it's a counter, if I put it inside the loop it'll reset to 0 at each iteration